sap/cli/startrfc.py

```diff
@@ -97,6 +104,22 @@ def startrfc(connection, args):
         return 1
 
     # rfc call passed, it is time for analysis of the returned response
+    response_formatted = FORMATTERS[args.output](resp)
+
+    # if the response file is given
+    if args.response_file:
+        try:
```

Comment:
What will happen if exception is thrown, but it is not of type (FileExistsError, IOError)? Wouldn't it be better to catch also some generic exception type to cover all cases?

Reply:
The program will crash with a stack trace. The builtin function open raises only these types in Python3. Of course, there could be RuntimeError but we usually do not catch these, also if I forgot to catch some type, I hope users will report or fix.

https://docs.python.org/3/library/functions.html#open - IOError is a symlink to OSError:

```
Changed in version 3.3:
The opener parameter was added.

The 'x' mode was added.

[IOError](https://docs.python.org/3/library/exceptions.html#IOError) used to be raised, it is now an alias of [OSError](https://docs.python.org/3/library/exceptions.html#OSError).

[FileExistsError](https://docs.python.org/3/library/exceptions.html#FileExistsError) is now raised if the file opened in exclusive creation mode ('x') already exists.
```